documentation/src/main/asciidoc/userguide/appendices/Configurations.adoc

```diff
@@ -79,6 +79,15 @@ Traditionally, Hibernate has considered the names locally scoped.
 If enabled, the names used by `@TableGenerator` and `@SequenceGenerator` will be considered global so configuring two different generators
 with the same name will cause a `java.lang.IllegalArgumentException` to be thrown at boot time.
 
+`*hibernate.jpa.compliance.criteria_copy*` (e.g. `true` (default value) or `false` )::
+The Jakarta Persistence spec says that mutations done to `CriteriaQuery`, `CriteriaUpdate` and `CriteriaDelete`
+after such objects were used to create a `jakarta.persistence.Query` may not affect that query.
+This requirement makes it necessary to copy these objects because the APIs allow mutations.
++
+If disabled, it is assumed that users do not mutate the criteria query afterwards
+and due to that, no copy will be created, which will improve performance.
+By default, the copy compliance is enabled because the Jakarta Persistence Criteria API is mutable.
```

Comment:
This is a tough one actually in terms of default.  I hate implicitly opting a user in to behavior we know will be worse performance.  Not to mention, this would be the first compliance setting we enable by default.

And really, how many people do this?  Create a criteria, use it to create a Query and then mutate the Criteria...